gitpod-io/gitpod
# Add metrics for content initialization

## Description
This adds metrics for initializers. Content initialization is either performed by ws-daemon (when object storage is used) or by supervisor (when PVC is used), so depending on the method the metric is either exposed by ws-daemon or supervisor. 

If ws-daemon performs the content init it runs as a container. The most straightforward way in this case to return the metrics to ws-daemon is to use the ready file which is written by the init process anyway and include the time and size there. Same approach is used by supervisor for consistency. It also means that you can now read `/workspace/.gitpod/ready` to "eyeball" what initializers where run and how long each one took when within your workspace.

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/14715

## How to test
- Start workspace
- Check /workspace/.gitpod/ready. It should contain the content size and the time it took the initializer to restore the content
- Port forward ws-daemon or ide-metrics
- curl localhost:9500/metrics

## Release Notes

```release-note
None
```

## Documentation


## Werft options:

      If enabled this will build `install/preview`
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
